cmd/cluster/status.go

```diff
@@ -0,0 +1,199 @@
+		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
+			Flags: flags,
+		},
+		IOStreams: streams,
+	}
+}
+
+func (o *statusOptions) complete(cmd *cobra.Command, _ []string) error {
+	var err error
+
+	k8svalid, err := o.k8sclusterresourcefactory.ValidateIdentifiers()
+	if !k8svalid {
+		if err != nil {
+			cmdutil.PrintErrorWithCauses(err, o.ErrOut)
+			return err
+		}
+
+	}
+
+	awsvalid, err := o.k8sclusterresourcefactory.Awscloudfactory.ValidateIdentifiers()
+	if !awsvalid {
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *statusOptions) run() error {
+
+	// This call gets the availability zone of the cluster, as well as the numbers of compute, infra, and master nodes.
+	az, compute, infra, master, err := ocmDescribe(o.k8sclusterresourcefactory.ClusterID)
+
+	nodesExpected := compute + infra + master
+
+	fmt.Fprintf(o.IOStreams.Out, "\nExpected number of nodes in availability zone(s) %s is : %v \n", az, nodesExpected)
+	fmt.Fprintf(o.IOStreams.Out, " \n")
+	if err != nil {
+		return err
+	}
+
+	// This aws client connects to an OpenShift AWS account and we use it here to get credentials to access a customer's account.
+	awsClient, err := o.k8sclusterresourcefactory.GetCloudProvider(o.verbose)
+	if err != nil {
+		return err
+	}
+
+	creds := o.k8sclusterresourcefactory.Awscloudfactory.Credentials
+
+	if o.k8sclusterresourcefactory.Awscloudfactory.RoleName != "OrganizationAccountAccessRole" {
+		creds, err = awsprovider.GetAssumeRoleCredentials(awsClient,
+			&o.k8sclusterresourcefactory.Awscloudfactory.ConsoleDuration, aws.String(o.k8sclusterresourcefactory.Awscloudfactory.SessionName),
+			aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s",
+				o.k8sclusterresourcefactory.AccountID,
+				o.k8sclusterresourcefactory.Awscloudfactory.RoleName)))
+		if err != nil {
+			klog.Error("Failed to assume BYOC role. Customer either deleted role or denied SREP access.")
+			return err
+		}
+	}
+
+	// Extracting region from the availability zone.
+	reg := az[0]
+	length := len(reg)
+	lastChar := reg[length-1 : length]
+	for _, r := range lastChar {
+		if unicode.IsLetter(r) {
+			reg = reg[0 : length-1]
+		}
+	}
+
+	//This call creates a client that is connected to the customer's account and we will use it to get the information on customer's running instances etc.
+	awsJumpClient, err := awsprovider.NewAwsClientWithInput(&awsprovider.AwsClientInput{
+		AccessKeyID:     *creds.AccessKeyId,
+		SecretAccessKey: *creds.SecretAccessKey,
+		SessionToken:    *creds.SessionToken,
+		Region:          reg,
+	})
+	if err != nil {
+		return err
+	}
+
+	log.Println("Getting instances info:")
+
+	instances, err := awsJumpClient.DescribeInstances(&ec2.DescribeInstancesInput{})
+
+	totalRunning := 0
+
+	//Here we count the number of customer's running instances in the region
+	for idx := range instances.Reservations {
+		for _, inst := range instances.Reservations[idx].Instances {
+			if *inst.State.Name == "running" {
```

Comment:
While we should also check for instances running, we should check the tags on a given instance to make sure that it's part of the cluster that we're looking at.  

With CCS we added the ability to add multiple clusters to the same account, and with BYOVPC customers could be running their own workloads on other instances within their cluster.

Reply:
Good point and good suggestion. I agree.

Reply:
I added the change.